vendor/github.com/weaveworks/common/middleware/grpc_logging.go

```diff
@@ -17,6 +17,11 @@ const (
 	errorKey = "err"
 )
 
+// If an error implements Observe(), it will get called and GRPCServerLog will do nothing.
+type Observer interface {
+	Observe(ctx context.Context, _ logging.Interface, method string, duration time.Duration)
```

Comment:
nit but seems weird to not give an argument in an interface a name
```suggestion
	Observe(ctx context.Context, log logging.Interface, method string, duration time.Duration)
```